### PR TITLE
feat: adds metadata generation for the solution

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

File: infra/metadata.yaml [hunk 1]
```diff
@@ -19,99 +19,113 @@ metadata:
...
+      - roles/compute.admin
+      - roles/compute.networkAdmin
+      - roles/firebase.managementServiceAgent
+      - roles/firebasehosting.admin
+      - roles/iam.serviceAccountAdmin
+      - roles/iam.serviceAccountUser
+      - roles/resourcemanager.projectIamAdmin
+      - roles/run.admin
+      - roles/secretmanager.admin
+      - roles/storage.admin
```
Comment (grayside) [inline infra/metadata.yaml:127]: note: I compared these to https://github.com/GoogleCloudPlatform/terraform-dynamic-python-webapp/blob/main/infra/test/setup/iam.tf#L17 and they appear to be the same. Seems like we will need to keep them manually in sync, unless we can parse the values from this config into the iam.tf.

File: infra/metadata.yaml [hunk 2]
```diff
@@ -19,99 +19,113 @@ metadata:
...
+    services:
+    - cloudresourcemanager.googleapis.com
+    - storage-api.googleapis.com
+    - serviceusage.googleapis.com
```
Comment (grayside) [inline infra/metadata.yaml:131]: question: I see only three services listed here, shouldn't this include all the services needed?

**Review by grayside — COMMENTED** [review]
Did some initial review but I'm still working to understand enough of the context.

File: infra/metadata.yaml (hunk 2 above)
Comment (glasnt) [reply infra/metadata.yaml:131]: I believe these are the services required to invoke terraform, which will then itself enable the other services. They're similar to the services [avocano gets you to manually enable before running the script](https://github.com/GoogleCloudPlatform/avocano/blob/main/README.walkthrough.md?plain=1#L28)

File: infra/metadata.yaml [hunk 3]
```diff
@@ -19,99 +19,113 @@ metadata:
...
+      varType: string
+      defaultValue: django
+    - name: database_username
+      description: Cloud SQL database name
+      varType: string
+      defaultValue: server
+    - name: image_version
+      description: Version of the Container Registry image to use
+      varType: string
+      defaultValue: v1.7.0
```
[the remark this answers is not in the document]
Comment (glasnt) [reply infra/metadata.yaml:65]: This is the avocano version, not the self-version. Updating this is usually manual, are you saying release-please can help with this overhead?

[the remark this answers is not in the document]
Comment (g-awmalik) [reply infra/metadata.yaml:65]: release-please won't help with this. This should updated in the TF variable when a newer version of the blueprint is released and the metadata will be updated accordingly.

File: infra/metadata.yaml [hunk 4]
```diff
@@ -19,99 +19,113 @@ metadata:
...
+      defaultValue: true
+    - name: instance_name
+      description: Cloud SQL Instance name
+      varType: string
+      defaultValue: psql
+    - name: labels
+      description: A set of key/value label pairs to assign to the resources deployed by this blueprint.
+      varType: map(string)
+      defaultValue: {}
+    - name: project_id
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply infra/metadata.yaml:78]: These are updated automatically with the TF variables defined for the blueprint.

File: infra/metadata.yaml (hunk 1 above)
Comment (g-awmalik) [reply infra/metadata.yaml:127]: infra/test/setup/iam.tf is the source for this as you rightly point out and will be updated automatically. I don't follow your comment "we will need to keep them manually in sync". Is there's another config you're keeping these roles in?

File: infra/metadata.yaml (hunk 2 above)
Comment (g-awmalik) [reply infra/metadata.yaml:131]: The source for this is the [`activate_apis`](https://github.com/GoogleCloudPlatform/terraform-dynamic-python-webapp/blob/main/infra/test/setup/main.tf#L27) variable to the project factory module under test/.

**Comment by g-awmalik** [comment]
> Did some initial review but I'm still working to understand enough of the context.

I somehow missed your comments from a week ago. Please let me know if anything is still unclear.

**Pull request merged**